h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -114,8 +116,9 @@ public static Frame merge(final Frame leftFrame, final Frame riteFrame, final in
     // and right in parallel was a little slower (97s) than one by one (89s).
     // empty frame will come back with base = Long.MIN_VALUE (-9223372036854775808).  
     // TODO: retest in future
-    RadixOrder leftIndex = createIndex(true ,leftFrame,leftCols,id_maps, ascendingL);
-    RadixOrder riteIndex = createIndex(false,rightFrame,riteCols,id_maps, ascendingR);
+    long randomDigits = System.nanoTime()+ new Random().nextLong();
```

Comment:
We can probably take the value from a unique sequence instead of using a timestamp. The risk of collision is low but there are better solutions.

eg. to have a static AtomicLong field on Merge class and call incrementAndGet() on this line. I can implement it if you want to.

Reply:
Yes, @michalkurka, can you please implement it and I will do the performance timing test?